parachain/runtime/runtime-tests/src/relaychain/kusama_cross_chain_transfer.rs

```diff
@@ -0,0 +1,449 @@
+        BuyExecution {
+            fees: (Parent, 96_000_000).into(),
+            weight_limit: Limited(expect_weight_limit),
+        },
+        DepositAsset {
+            assets: All.into(),
+            max_assets: 1,
+            beneficiary: Here.into(),
+        },
+    ]);
+    Kintsugi::execute_with(|| {
+        let r = XcmExecutor::<XcmConfig>::execute_xcm(Parent, message, expect_weight_limit);
+        assert_eq!(r, Outcome::Complete(expect_weight_limit));
+    });
+}
+
+#[test]
+fn subscribe_version_notify_works() {
+    // relay chain subscribe version notify of para chain
+    KusamaNet::execute_with(|| {
+        let r = pallet_xcm::Pallet::<kusama_runtime::Runtime>::force_subscribe_version_notify(
+            kusama_runtime::Origin::root(),
+            Box::new(Parachain(2092).into().into()),
+        );
+        assert_ok!(r);
+    });
+    KusamaNet::execute_with(|| {
+        kusama_runtime::System::assert_has_event(kusama_runtime::Event::XcmPallet(
+            pallet_xcm::Event::SupportedVersionChanged(
+                MultiLocation {
+                    parents: 0,
+                    interior: X1(Parachain(2092)),
+                },
+                2,
+            ),
+        ));
+    });
+
+    // para chain subscribe version notify of relay chain
+    Kintsugi::execute_with(|| {
+        let r = pallet_xcm::Pallet::<kintsugi_runtime_parachain::Runtime>::force_subscribe_version_notify(
+            Origin::root(),
+            Box::new(Parent.into()),
+        );
+        assert_ok!(r);
+    });
+    Kintsugi::execute_with(|| {
+        System::assert_has_event(kintsugi_runtime_parachain::Event::PolkadotXcm(
+            pallet_xcm::Event::SupportedVersionChanged(
+                MultiLocation {
+                    parents: 1,
+                    interior: Here,
+                },
+                2,
+            ),
+        ));
+    });
+
+    // para chain subscribe version notify of sibling chain
+    Kintsugi::execute_with(|| {
+        let r = pallet_xcm::Pallet::<kintsugi_runtime_parachain::Runtime>::force_subscribe_version_notify(
+            Origin::root(),
+            Box::new((Parent, Parachain(2001)).into()),
+        );
+        assert_ok!(r);
+    });
+    Kintsugi::execute_with(|| {
+        assert!(kintsugi_runtime_parachain::System::events().iter().any(|r| matches!(
+            r.event,
+            kintsugi_runtime_parachain::Event::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent(Some(_)))
+        )));
+    });
+    Sibling::execute_with(|| {
+        assert!(testnet_runtime_parachain::System::events().iter().any(|r| matches!(
+            r.event,
+            testnet_runtime_parachain::Event::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent(Some(_)))
+                | testnet_runtime_parachain::Event::XcmpQueue(cumulus_pallet_xcmp_queue::Event::Success(Some(_)))
+        )));
+    });
+}
+
+#[test]
+fn trap_assets_works() {
+    TestNet::reset();
```

Comment:
Why do we need this? Can you move this to a shared init routine and maybe add a comment?